scm-step/src/test/java/org/jenkinsci/plugins/workflow/steps/scm/GitStepTest.java

```diff
@@ -0,0 +1,31 @@
+package org.jenkinsci.plugins.workflow.steps.scm;
+
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepConfigTester;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+/**
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+public class GitStepTest {
+
+    @Rule public JenkinsRule r = new JenkinsRule();
+
+    @Test
+    public void roundtrip() throws Exception {
+        GitStep step = new GitStep("git@github.com:jenkinsci/workflow-plugin.git");
+        Step roudtrip = new StepConfigTester(r).configRoundTrip(step);
+        r.assertEqualDataBoundBeans(step, roudtrip);
+    }
+
+    @Test
+    public void roundtrip_withcredentials() throws Exception {
+        GitStep step = new GitStep("git@github.com:jenkinsci/workflow-plugin.git");
+        step.setCredentialsId("1234567890abcdef");
+        Step roudtrip = new StepConfigTester(r).configRoundTrip(step);
+        r.assertEqualDataBoundBeans(step, roudtrip);
```

Comment:
Test results are not in yet, but I cannot imagine how this could pass! You have not edited `GitStep/config.jelly` to add an entry with `<c:select/>`, nor created `GitStep.DescriptorImpl.doFillCredentialsIdItems` to enumerate the available credentials.

And of course the test would need to actually create a credentials item of a suitable type: a dummy `credentialsId` would not appear in the pulldown and so could not be round-tripped. Note that the actual Snippet Generator does not ever round-trip anything, but `StepConfigTester` is nonetheless helpful in ensuring that all required components of form databinding are present and functioning.